### DISCUSSION TITLE
Perform an action only when a testcase fails

### DISCUSSION THREAD

**Discussion by bhanu-pratap** [xunit/xunit#2211] [Question] [answered]
I am new to xinit.net and I was looking for a way to identify at runtime test status and perform some action only when the test fails. Although in other test frameworks it can be done by using test context or listeners , I am looking for a way to achive it with xunit.net.
Any help would be appreciated.

**Comment by dtopuzov** [comment]
I am also looking for such options.
Take screenshots on test fail is common scenario in e2e ui testing.

**Comment by bradwilson** [comment, marked as answer]
There is nothing in v2 for this, but it's on the [v3 roadmap](https://github.com/xunit/xunit/issues/2133): ["Add test context object"](https://github.com/xunit/xunit/issues/621)

This "open for discussion" item may also apply: ["Introduce Snapshot Testing Like Jest"](https://github.com/xunit/xunit/issues/1727)

**Reply by bradwilson** [reply]
Going to mark this as answered. Please let me know if you have any further questions.
